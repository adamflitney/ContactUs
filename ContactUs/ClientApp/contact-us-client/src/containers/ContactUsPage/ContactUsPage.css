.contact-us {
    height: 100vh;
    display: grid;
    place-items: center;
}

.form-wrapper {
    max-width: 800px;
    margin: 0 auto;
    width: 100%;
    position: relative;
}

h1 {
    margin-bottom: 3em;
    font-size: 5em;
    text-align: center;
}

.alert {
    position: absolute;
    bottom: -100px;
    text-align: center;
}