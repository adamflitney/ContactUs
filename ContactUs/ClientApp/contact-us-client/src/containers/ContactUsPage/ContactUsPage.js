import React, { useState } from 'react';
import { Fade } from 'react-bootstrap';
import './ContactUsPage.css';
import ContactForm from '../../components/ContactForm/ContactForm';
import postMessage from '../../api/ContactUsAPI';

const ContactUsPage = props => {

  const [success, setSuccess] = useState("");

  const handleSubmit = async (data) => {
    const result = await postMessage(data);
    if(result.success && result.success === "true"){
      setSuccess("true");
    setTimeout(() => {
      setSuccess("")
    }, 3000);
    } else {
      setSuccess("false");
    setTimeout(() => {
      setSuccess("")
    }, 3000);
    }
    
    console.log(result);
  }

  return (
    <div className="contact-us">

      <div className="form-wrapper">
        <h1>Contact Me!</h1>
        <ContactForm submitHandler={handleSubmit} />
        <Fade in={success === "true"}>
        <div className="alert alert-dismissible alert-success">
          Message sent successfully.
        </div>
        </Fade>
        <Fade in={success === "false"}>
        <div className="alert alert-dismissible alert-danger">
          Message failed to send.
        </div>
        </Fade>
        
      </div>

    </div>
  );
}

export default ContactUsPage;