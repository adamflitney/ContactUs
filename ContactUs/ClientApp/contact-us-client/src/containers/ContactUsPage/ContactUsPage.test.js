import React from 'react';
import renderer from 'react-test-renderer';
import ContactUsPage from './ContactUsPage';
import ContactUsForm from '../../components/ContactForm/ContactForm';

describe('ContactUsPage', () => {
    const page = renderer.create(<ContactUsPage />);
    it('renders correctly', () => {
        const tree = page.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the contact form', () => {
        const pageInstance = page.root;
        expect(pageInstance.findByType(ContactUsForm)).toBeTruthy();
    })
})