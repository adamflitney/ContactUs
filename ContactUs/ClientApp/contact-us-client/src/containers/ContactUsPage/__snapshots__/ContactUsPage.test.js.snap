// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUsPage renders correctly 1`] = `
<div
  className="contact-us"
>
  <div
    className="form-wrapper"
  >
    <h1>
      Contact Me!
    </h1>
    <form
      className="contact-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="contactFormName"
        >
          Name
        </label>
        <input
          className="form-control"
          id="contactFormName"
          name="name"
          onChange={[Function]}
          placeholder="e.g. Joe Bloggs"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="contactFormEmail"
        >
          Email Address
        </label>
        <input
          className="form-control"
          id="contactFormEmail"
          name="email"
          onChange={[Function]}
          placeholder="e.g. joe@bloggs.com"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="contactFormMessage"
        >
          Enter your message
        </label>
        <textarea
          className="form-control"
          id="contactFormMessage"
          name="message"
          onChange={[Function]}
          placeholder="Enter your message here."
          required={true}
          rows="4"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-success"
        id="submit-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      className="fade alert alert-dismissible alert-success"
    >
      Message sent successfully.
    </div>
    <div
      className="fade alert alert-dismissible alert-danger"
    >
      Message failed to send.
    </div>
  </div>
</div>
`;
